utils/obsproc/IcecAmsr2Ioda.h

```diff
@@ -66,18 +69,31 @@ namespace gdasapp {
       iodaVars.referenceDate_ = "seconds since 1970-01-01T00:00:00Z";
 
       size_t index = 0;
-      std::tm timeinfo = {};
       for (int i = 0; i < ntimes; i += dimTimeSize) {
-        timeinfo.tm_year = oneTmpdateTimeVal[i] - 1900;  // Year since 1900
-        timeinfo.tm_mon = oneTmpdateTimeVal[i + 1] - 1;  // 0-based; 8 represents Sep
-        timeinfo.tm_mday = oneTmpdateTimeVal[i + 2];
-        timeinfo.tm_hour = oneTmpdateTimeVal[i + 3];
-        timeinfo.tm_min = oneTmpdateTimeVal[i + 4];
-        timeinfo.tm_sec = oneTmpdateTimeVal[i + 5];
-
-        // Calculate and store the seconds since the Unix epoch
-        time_t epochtime = std::mktime(&timeinfo);
-        iodaVars.datetime_(index) = static_cast<int64_t>(epochtime);
+        int year = oneTmpdateTimeVal[i];
+        int month = oneTmpdateTimeVal[i+1];
+        int day = oneTmpdateTimeVal[i+2];
+        int hour = oneTmpdateTimeVal[i+3];
+        int minute =  oneTmpdateTimeVal[i+4];
+        int second = static_cast<int>(oneTmpdateTimeVal[i+5]);
+
+        // Bypassing validYYYYMMDD function within dateToJulian
+        if (year == -9999 && month == -9999 && day == -9999 &&
+             hour == -9999 && minute == -9999 && second == -9999) {
+          year = 0, month = 0, day = 0;
+          hour = 0, minute = 0, second = 0;
+        }
+
+        uint64_t julianDate = util::datefunctions::dateToJulian(year, month, day);
+
+        // Convert a date to Julian date
+        // Subtract Julian date for January 1, 1970 (epoch)

```

Comment:
check English in the comment ... My frenglish tells me it's wrong.

Reply:
> check English in the comment ... My frenglish tells me it's wrong.

OMG, yes, I will correct it.

Reply:
Change it to:
"Subtract Julian date from January 1, 1970 (convert to epoch)"